### DISCUSSION TITLE
when is eepitch-kill really needed  ?

### DISCUSSION THREAD

**Discussion by martin12333** [edrx/eev#4] [Q&A] [answered]
I am still not sure, when is it really needed to run the line

```
 (eepitch-kill)

```

... Maybe when I want to demo something publicly, and i want to clear the mess, that was in the terminal before the demo starts ?

(I have made some e-scripts, and I notice, I  copied only the line:

```
 (eepitch-eshell)
```
)

(i am a beginner in Emacs)

**Comment by edrx** [comment, marked as answer]
Running the (eepitch-kill) is only needed when you want to make sure that you are in a new shell, or in a new REPL of another kind. In most cases I run f8s in the three lines of an eepitch block to make sure that I'm in a new REPL... the most common case in which I run just an (eepitch-shell) - or an (eepitch-shell2), or an (eepitch-shell3) - is when I'm using the shell to download something that takes a long time.

Btw, I'm going to spend a lot of time in the #emacs IRC channel today and tomorrow, and I have a lot of material for beginners, but it's all scattered around... did you see this?

(find-rcirc-intro)

http://anggtwu.net/eev-intros/find-rcirc-intro.html

**Reply by martin12333** [reply]
(i am learning now how to use irc for the first time)

(i am reading and using M-e now, in my  Emacs,
at my	~/OneDrive/emacs/find-rcirc-intro.f8.el,
at the first

```
  (find-libera-3a "#eev")
```
)

(i have to take a break now, i will continue in less than 30 minutes)

**Reply by martin12333** [reply]
but

> Public logging of the Emacs channel is forbidden.

; https://www.emacswiki.org/emacs/EmacsChannel

(so I will probably prefer Github to IRC)

**Comment by edrx** [comment]
Well, public logging of the #eev channel is allowed! =)
